### PR TITLE
Transformation change datatype

### PR DESCRIPTION
**Pull request by richadivami**
# TITLE

## Description
Summary of changes

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)

### CODE REVIEW

**Review by adityadivami — CHANGES_REQUESTED** [review]
CR-1. Use i18n wherever we have labels in the DOM and reuse same if required in the states, utils and services
CR-2. Variable names should be camelCase
CR-3. Each file should have copyrights text
CR-4. Any file can contain only 1 component
CR-5. Use export default function () {} for declaring any component
CR-6. Add proper null checks
CR-7. Add interfaces to states, props, variables, functions. Never use any as type.
CR-8. Spread props as component arguments, rather than using props directly
CR-9: Remove commented code and console logs
CR-10: Make sure the branch has changes only specific to the feature or the task it is expected
CR-11: Use Typography wherever we have labels being displayed on the screen. Do not write text inside div or p tags directly.

//ignore for now
CR-12: Do not keep logic / data manipulation / API calls in the component file. Move logic / data manipulation to utils and API calls to services. Create folder for services and keep function name as the filename for each service.
CR-13. Add Data Testids to the following items
a. onClick, onChange and any event triggering elements
b. All labels whether Typograpahy or not

File: app/cdap/components/GridTable/index.tsx [hunk 1]
```diff
@@ -52,21 +58,23 @@ export default function GridTable() {
 
   const classes = useStyles();
 
-  const [loading, setLoading] = useState(false);
+  const [loading, setLoading] = useState<boolean>(false);
   const [headersNamesList, setHeadersNamesList] = useState([] as IHeaderNamesList[]);
-  const [rowsDataList, setRowsDataList] = useState([]);
+  const [rowsDataList, setRowsDataList] = useState([] as IRowData[]);
```
Comment (adityadivami) [inline app/cdap/components/GridTable/index.tsx]: const [rowsDataList, setRowsDataList] = useState([] as IRowData[]);

should be 
const [rowsDataList, setRowsDataList] = useState<IRowData[]>([]);

Comment (adityadivami) [reply app/cdap/components/GridTable/index.tsx]: Please make sure all types are given like this

File: app/cdap/components/AddTransformation/utils.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import { DATATYPE_OPTIONS } from '../GridTable/components/NestedMenu/menuOptions/datatypeOptions';
+export const getDirective = (functionName: string, columnSelected: string) => {
+  if (DATATYPE_OPTIONS.some((item) => item.value === functionName)) {
```
Comment (dlakshmi123) [inline app/cdap/components/AddTransformation/utils.ts:18]: null check

File: app/cdap/components/AddTransformation/index.tsx [hunk 3]
```diff
@@ -62,7 +63,8 @@ export default function({
   };
 
   const handleApply = () => {
-    applyTransformation('');
+    const directive = getDirective(functionName, selectedColumns[0].label);
```
Comment (dlakshmi123) [inline app/cdap/components/AddTransformation/index.tsx:66]: add type for directive

**Comment by dlakshmi123** [comment]
CR-1. Add Data Testids to the following items
a. onClick, onChange and any event triggering elements
b. All labels whether Typograpahy or not
CR-2. Use i18n wherever we have labels in the DOM and reuse same if required in the states, utils and services
CR-3. Variable names should be camelCase
CR-4. Each file should have copyrights text
CR-5. Any file can contain only 1 component
CR-6. Use export default function () {} for declaring any component
CR-7. Add proper null checks
CR-8. Add interfaces to states, props, variables, functions. Never use any as type.
CR-9. Spread props as component arguments, rather than using props directly
CR-10: Remove commented code and console logs
CR-11: Make sure the branch has changes only specific to the feature or the task it is expected
CR-12: Use Typography wherever we have labels being displayed on the screen. Do not write text inside div or p tags directly.

//ignore for now
CR-13: Do not keep logic / data manipulation / API calls in the component file. Move logic / data manipulation to utils and API calls to services. Create folder for services and keep function name as the filename for each service.

File: app/cdap/components/GridTable/index.tsx [hunk 4]
```diff
@@ -42,9 +42,13 @@ import {
...
   IMissingListData,
+  IGridParams,
+  IRequestBody,
+  IApiPayload,
 } from './types';
 import { convertNonNullPercent } from './utils';
 import AddTransformation from 'components/AddTransformation';
+import { getAPIRequestPayload } from './services';
 
 export default function GridTable() {
```
Comment (dlakshmi123) [inline app/cdap/components/GridTable/index.tsx:53]: use anonymous function

File: app/cdap/components/AddTransformation/index.tsx (hunk 3 above)
Comment (richadivami) [reply app/cdap/components/AddTransformation/index.tsx:66]: Done

File: app/cdap/components/AddTransformation/utils.ts (hunk 2 above)
Comment (richadivami) [reply app/cdap/components/AddTransformation/utils.ts:18]: Done

File: app/cdap/components/GridTable/index.tsx (hunk 4 above)
Comment (richadivami) [reply app/cdap/components/GridTable/index.tsx:53]: Done

**Pull request merged**